### PR TITLE
Feature/search trip

### PR DESCRIPTION
**Pull request by JorisPineau**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [ ] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by herve-perel — APPROVED** [review]

**Review by Abecker37 — APPROVED** [review]

File: packages/frontend/web/src/pages/SearchTrip.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+          placeEnd?.x
+        }&endY=${placeEnd?.y}&passenger=${data.passenger}&date=${date}`,
+      ).then((res) => res.json());
+      setSearchTripFilter(response);
+    } catch (error) {
+      throw new Error(String(error));
+    }
+  };
+
+  const passengers = [1, 2, 3, 4, 5, 6, 7, 8];
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/SearchTrip.tsx:145]: why, pourquoi est-ce ici ? Que sont ces séquences de chiffres à la fin ??

File: packages/frontend/web/src/pages/SearchTrip.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+  return (
+    <div className='flex justify-center'>
+      <div className=' text-primary flex h-[750px] w-[600px] justify-center bg-black'>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/SearchTrip.tsx:149]: espace vide

File: packages/frontend/web/src/pages/SearchTrip.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+import { useAutocomplete } from '@vis.gl/react-google-maps';
+import { useRef, useState } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Controller, useForm } from 'react-hook-form';
+
+import {
+  type SearchTripType,
+  searchTripSchema,
+} from '@app/types/src/search-trip-validation';
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/SearchTrip.tsx:11]: mauvais import

File: packages/frontend/web/src/components/CardTrip.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          new Date(search.date).getMinutes() + search.cp_t_travel_time,
+        );
+        const init = new Date().setHours(0, 0, 0, 0);
+        const travelTime = new Date(init).setHours(
+          new Date(init).getHours(),
+          new Date(init).getMinutes() + search.cp_t_travel_time,
+        );
+        const timeEndFormat = `${new Date(timeEnd).getHours().toString().padStart(2, '0')}h${new Date(timeEnd).getMinutes().toString().padStart(2, '0')}`;
+        const timeStartFormat = `${new Date(timeStart).getHours().toString().padStart(2, '0')}h${new Date(timeStart).getMinutes().toString().padStart(2, '0')}`;
+        const travelTimeFormat = `${new Date(travelTime).getHours().toString().padStart(2, '0')}h${new Date(travelTime).getMinutes().toString().padStart(2, '0')}`;
```
Comment (YDrogen) [inline packages/frontend/web/src/components/CardTrip.tsx:52]: Ça peut sortir du return ça il me semble
Y'a du code en double aussi j'ai l'impression 🌮

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by Abecker37 — DISMISSED** [review]

**Review by herve-perel — DISMISSED** [review]

File: packages/frontend/web/package.json [hunk 5]
```diff
@@ -30,6 +28,8 @@
     "postcss": "^8.4.31",
     "tailwindcss": "^3.3.5",
     "typescript": "^5.0.2",
-    "vite": "^4.4.5"
+    "vite": "^4.4.5",
+    "@types/react-datepicker": "^4.19.5",
+    "@hookform/resolvers": "^3.3.4"
```
Comment (YDrogen) [inline packages/frontend/web/package.json:33]: C'est pas un type ça :(

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by herve-perel — DISMISSED** [review]

**Review by herve-perel — APPROVED** [review]

File: packages/types/src/index.ts [hunk 6]
```diff
@@ -1,5 +1,7 @@
 export * from './database';
 export * from './auth';
+export * from '@app/shared/src/search-trip-validation';
```
Comment (YDrogen) [inline packages/types/src/index.ts:3]: mauvais chemin

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**